### PR TITLE
Bug: ProductsPuller sometimes creates duplicate SKUs

### PR DESCRIPTION
**Pull request by jmataya**
- [x] Add tests for the case when duplicate SKUs are created
  - This happens when an Admin has created the master in the site Admin before the import is started
- [x] Add check for master SKU before deciding when to attach to a master
- [x] Get tests to be green

### CODE REVIEW

File: app/async/workers/products_puller.rb [hunk 1]
```diff
@@ -166,11 +174,11 @@ def update_variant(product, spree_variant, log, shipping_category_id, item_categ
       spree_variant.product.save!
     end
 
-    def attach_to_master(product, spree_variants, item_category_id, log)
+    def attach_to_master(product, item_category_id, log)
       ## build a master variant sku which would be the same color code with 0000
       product_code = product['sku'].match(/^(.*)-/)[1].to_s
       master_variant_sku = "#{product_code}-00"
-      master_variant = spree_variants.find { |variant| variant.sku == master_variant_sku && variant.is_master }
+      master_variant = Spree::Variant.find { |variant| variant.sku == master_variant_sku && variant.is_master }
```
Comment (Axblade) [inline app/async/workers/products_puller.rb]: Won't it be better in terms of performance to get all master variants first and then iterate through an array to find needed? Or may be even use a hash of master-skus and master variants?

Comment (Axblade) [inline app/async/workers/products_puller.rb]: Ah, yes. We are creating new master variants all the time while syncing products. Your variant is correct. :+1:

**Comment by Axblade** [comment]
Looks good
